### ISSUE TITLE
Make pyrosm compatible with newer numpy and shapely version.

### ISSUE THREAD

**Issue by LuposX** [HTenkanen/pyrosm#208]
**Describe the bug**
When trying to run:
```
osm = pyo.OSM("guernsey-jersey-latest.osm.pbf")
buildings = osm.get_buildings()
```
I get the error:
```
AttributeError: module 'shapely.geos' has no attribute 'lgeos'
```

**To Reproduce**
1. create new conda enviroment 3.8
2. conda install Pyrosm
3. conda install jupyter lab
4. download "andorra-latest.osm.pbf" from: https://download.geofabrik.de/europe.html , just an exampel didnt work with any .pbf file.
5. run the commands described above

**Expected behavior**
No error, it loads all buildings in the region

**Environment:**
 - OS: Pop!_OS 22.04 LTS
 - Python package source: conda
 - Versions of Python: 3.8

**Test data and/or script/notebook**
[requirements.txt](https://github.com/HTenkanen/pyrosm/files/10272112/requirements.txt)
[notebook_and_data.zip](https://github.com/HTenkanen/pyrosm/files/10272118/notebook_and_data.zip)

**Comment by LuposX** [comment]
Fixed after downgrading to shapely=1.8.5
May have sth. to do with #191

**Comment by momince** [comment]
I have installed shapely=1.8.5
now my kernel crashed. it happens again and again.
Can you please share how to build a completely new environment and the package version to avoid this "module 'shapely.geos' has no attribute 'lgeos' " issue? 
Thanks.

**Comment by LuposX** [comment]
I removed shapely 2.0 and downgraded to shapely=1.8.5 and I didnt get said issue anymore, If the data I work with is very big I get the expception: 
`AttributeError: module 'numpy' has no attribute 'float'`, though.

You can try to use my attachment file, to create an envireoment, with the command `conda env create --file requirement.yml` it should work.
[requirement.zip](https://github.com/HTenkanen/pyrosm/files/10288373/requirement.zip)

**Comment by momince** [comment]
Thank you so much. It worked. And to solve the problem,
"AttributeError: module 'numpy' has no attribute 'float'"

please use this line in your terminal 
>pip install "numpy<1.24"

**Comment by LuposX** [comment]
Thanks for the tipp.
 I think it worked, I am not sure I will keep an eye on it and see if the error occurs again.

**Comment by Geometrein** [comment]
Is anyone  else experiencing the same:

```AttributeError: module 'shapely.geos' has no attribute 'lgeos'```
&
```"AttributeError: module 'numpy' has no attribute 'float'"```

errors even when using  ```shapely=1.8.5```& ```numpy=1.24```?

**Comment by tfardet** [comment]
Hi there, following up on this issue, are there plans to upgrade ``pyrosm`` to move to ``shapely>2`` in the near future?
Of course, as this means rewriting the code to stop using ``pygeos``, I understand that this is not exactly a minor change.

**Comment by HTenkanen** [comment]
Hi all! And thanks for active discussion and helping each other out!

@tfardet : There is a plan to start supporting shapely 2.0 and in general have a better memory management for the library. Unfortunately, I've been extremely busy at my day job and haven't found time for pyrosm. Hopefully, this situation will improve now during the coming months. However, all contributions to the library are also more than welcome to more quickly advance with things. 😉

**Comment by tfardet** [comment]
@HTenkanen thanks for the quick answer!
I completely understand, I would also love to help but I can't spare the time it would require...
Good to know that things might move forward in the future, though, good luck and thanks for your work!

**Comment by snoop168** [comment]
Was wondering if there was any movement on this issue. I'm having issues being able to run on any python version 3.8 or higher. No matter what I do it doesnt work. I install with conda-forge and then downgrade both numpy and shapely as mentioned in this issue but I still run into issues. After downgrading I receive this warning 
```The Shapely GEOS version (3.10.3-CAPI-1.16.1) is incompatible with the GEOS version PyGEOS was compiled with (3.11.2-CAPI-1.17.2). Conversions between both will be slow.```

If I continue anyway, when running ```get_data_by_custom_criteria(custom_filter=)``` I get a message saying the kernel crashed and will be automatically restarted. I am using Jupyter Labs.

Any update, or help would be appreciated. I was going to try to use Python 3.6 but it looks like a hassle to get Anaconda to create a venv with anything less than 3.8...

**Comment by anastassiavybornova** [comment]
> I removed shapely 2.0 and downgraded to shapely=1.8.5 and I didnt get said issue anymore, If the data I work with is very big I get the expception: `AttributeError: module 'numpy' has no attribute 'float'`, though.
> 
> You can try to use my attachment file, to create an envireoment, with the command `conda env create --file requirement.yml` it should work. [requirement.zip](https://github.com/HTenkanen/pyrosm/files/10288373/requirement.zip)

For me this suggested solution worked with the additional step of switching off pygeos use (`os.environ['USE_PYGEOS'] = '0'`) when importing the packages, like so:

first, in the terminal

```
conda env create -f requirement.yml 
conda activate PSE
```

and then in the notebook (commented out = versions that in my case were installed in the PSE env)

```
import pyrosm # 0.6.1
import numpy # 1.23.5
import shapely # 1.8.5.post1
import pandas # 2.0.3
import geopandas # 0.13.2
import networkx # 3.1
import os
os.environ['USE_PYGEOS'] = '0'
```

**Comment by MapMaker2023** [comment]
First thanks & kudos to @HTenkanen for such a useful package. Actively using it for research on [this](https://dataportal.arc.gov.au/NCGP/Web/Grant/Grant/LE220100028) project. The ability to create and save Pandana graphs is particular helpful.

Second, depreciating numpy to 1.19.5 after a ```conda install -c conda-forge pyrosm``` gets passed the ```AttributeError: module 'numpy' has no attribute 'float'``` for me.

**Comment by HTenkanen** [comment]
This should be now fixed in the new release pyrosm v0.6.2 which fixes the numpy related issues. See #214
